File: Server/app/routes/category.go
package routes

import (
	"DB-Server/app/handles"
	"DB-Server/app/middlewares"

	"github.com/gin-gonic/gin"
)

func categoryRoute(r *gin.Engine) {
	router := r.Group("/category")
	// 添加分类
	router.POST("/add", middlewares.AdminToken(), handles.AddCate)
	// 更新分类
	router.POST("/update", middlewares.AdminToken(), handles.UpdateCate)
	// 删除分类
	router.POST("/delete", middlewares.AdminToken(), handles.DeleteCate)
	// 分类列表
	router.GET("/list", handles.GetCateList)
}

File: client/src/api/category.js
import request from '../utils/request'
import qs from 'qs'

// 分类列表
export function CateList() {
    return request({
        url: '/category/list',
        method: 'GET'
    })
}

// 添加分类
export function AddCate(data) {
    return request({
        url: '/category/add',
        method: 'POST',
        data: data
    })
}

// 更新分类
export function UpdateCate(data) {
    return request({
        url: '/category/update',
        method: 'POST',
        data: data
    })
}

// 删除分类
export function DeleteCate(data) {
    return request({
        url: '/category/delete',
        method: 'POST',
        data: qs.stringify(data)
    })
}